suricata/update/main.py

```diff
@@ -351,6 +351,9 @@ def get_tmp_filename(self, url):
             "%s-%s" % (url_hash, self.url_basename(url)))
 
     def fetch(self, url):
+        net_arg = url
+        if isinstance(url, tuple):
+            url = url[0]
```

Comment:
I'm not sure if I like the idea of a tuple here. It works, and perhaps we can leave it for now.  But what if we were to allow for http basic authentication as well, and we had to pass in a username and password?  Would you do you this differently?

I'm pretty sure http basic auth will come as a feature request sooner or later, so I think its a good example for future growth in the arguments passed here.

Reply:
Yes, that would be the case.